src/App.tsx

```diff
@@ -1,39 +1,124 @@
+//   const value = Math.random().toString().slice(2, 6);
+
+//   return `Clock-${value}`;
+// }
+
+// type State = {
+//   hasClock: boolean,
+//   clockName: string,
+// };
+
+// export class App extends Component<{}, State> {
+//   state = {
+//     hasClock: true,
+//     clockName: getRandomName(),
+//   };
+
+//   intervalName = 0;
+
+//   componentDidMount() {
+//     window.addEventListener('contextmenu', () => {
+//       this.setState({
+//         hasClock: false,
+//       });
+//     });
+
+//     document.addEventListener('click', () => {
+//       this.setState({
+//         hasClock: true,
+//       });
+//     });
+
+//     this.intervalName = window.setInterval(() => {
+//       this.setState({
+//         clockName: getRandomName(),
+//       });
+//     }, 3300);
+
+//     if (!this.state.hasClock) {
+//       window.clearInterval(this.intervalName);
+//     }
+//   }
+
+//   render() {
+//     const { hasClock, clockName } = this.state;
+
+//     return (
+//       <div className="App">
+//         <h1>React clock</h1>
+//         {hasClock && <Clock clockName={clockName} />}
+//       </div>
+//     );
+//   }
+// }
+
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+  const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export interface State {
+  hasClock: boolean;
+  clockName: string;
+}
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: false,
+    clockName: getRandomName(),
+  };
+
+  date = new Date();
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  clockNameTimer = 0;
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidMount() {
+    this.setState({ hasClock: true });
+    this.clockNameTimer = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.addEventListener('contextmenu', (event: Event) => {
+      event.preventDefault();
+      this.setState({ hasClock: false });
+    });
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    document.addEventListener('click', () => (
+      this.setState({ hasClock: true })
+    ));
+  }
 
-        {' time is '}
+  componentWillUnmount() {
+    clearInterval(this.clockNameTimer);
+    this.setState({ hasClock: false });
+  }
 
-        <span className="Clock__time">
-          {today.toLocaleTimeString()}
-        </span>
+  changeClockVisibility() {
+    const { hasClock } = this.state;
+
+    return this.setState({ hasClock: !hasClock });
+  }
```

Comment:
this method is reduntant you can just pass callback as an argument to the setState method